### PR TITLE
Better Auto0 headers handling

### PR DESCRIPTION
**Pull request by gilgardosh**
Handle Fetch API requests/headers

### CODE REVIEW

File: packages/plugins/auth0/src/index.ts [hunk 1]
```diff
@@ -58,20 +58,27 @@ export const useAuth0 = <TOptions extends Auth0PluginOptions>(options: TOptions)
           `useAuth0 plugin unable to locate your request or headers on the execution context. Please make sure to pass that, or provide custom "extractTokenFn" function.`
         );
       } else {
+        let authHeader = '';
         if (headers[headerName] && typeof headers[headerName] === 'string') {
-          const authHeader = headers[headerName] || '';
-          const split = authHeader.split(' ');
+          authHeader = headers[headerName] || '';
+        } else if (headers.get && headers.get(headerName)) {
```
Comment (dotansimha) [inline packages/plugins/auth0/src/index.ts]: Use `headers.has` here

File: packages/plugins/auth0/src/index.ts [hunk 2]
```diff
@@ -58,20 +58,27 @@ export const useAuth0 = <TOptions extends Auth0PluginOptions>(options: TOptions)
           `useAuth0 plugin unable to locate your request or headers on the execution context. Please make sure to pass that, or provide custom "extractTokenFn" function.`
         );
       } else {
+        let authHeader = '';
```
Comment (dotansimha) [inline packages/plugins/auth0/src/index.ts]: default maybe should be `null`?

Comment (dotansimha) [reply packages/plugins/auth0/src/index.ts]: add strict type please `authHeader: string | null`

File: packages/plugins/auth0/src/index.ts [hunk 3]
```diff
@@ -58,20 +58,27 @@ export const useAuth0 = <TOptions extends Auth0PluginOptions>(options: TOptions)
           `useAuth0 plugin unable to locate your request or headers on the execution context. Please make sure to pass that, or provide custom "extractTokenFn" function.`
         );
       } else {
+        let authHeader: string | null = null;
         if (headers[headerName] && typeof headers[headerName] === 'string') {
-          const authHeader = headers[headerName] || '';
-          const split = authHeader.split(' ');
+          authHeader = headers[headerName] || null;
+        } else if (headers.get && headers.has(headerName)) {
```
Comment (dotansimha) [inline packages/plugins/auth0/src/index.ts]: `} else if (headers.get && headers.has && headers.has(headerName)) {`

File: packages/plugins/auth0/src/index.ts [hunk 4]
```diff
@@ -58,20 +58,27 @@ export const useAuth0 = <TOptions extends Auth0PluginOptions>(options: TOptions)
...
-            const [type, value] = split;
+        const split = authHeader.split(' ');
 
-            if (type !== tokenType) {
-              throw new Error(`Unsupported token type provided: ${type}!`);
-            } else {
-              return value;
-            }
+        if (split.length !== 2) {
+          throw new Error(`Invalid value provided for header "${headerName}"!`);
```
Comment (dotansimha) [inline packages/plugins/auth0/src/index.ts]: I wonder if we need to throw here `EnvelopError`?

File: packages/plugins/auth0/src/index.ts [hunk 5]
```diff
@@ -58,20 +58,27 @@ export const useAuth0 = <TOptions extends Auth0PluginOptions>(options: TOptions)
...
+        } else {
+          const [type, value] = split;
+
+          if (type !== tokenType) {
+            throw new Error(`Unsupported token type provided: ${type}!`);
```
Comment (dotansimha) [inline packages/plugins/auth0/src/index.ts]: same, throw here `EnvelopError`?

File: packages/core/test/plugins/use-masked-errors.spec.ts [hunk 6]
```diff
@@ -319,4 +320,28 @@ Array [
...
+    expect.assertions(2);
+    const auto0Options = {
+      domain: 'domain.com',
+      audience: 'audience',
+      headerName: 'authorization',
+      preventUnauthenticatedAccess: false,
+      extendContextField: 'auth0',
+      tokenType: 'Bearer',
+    };
+    const testInstance = createTestkit([useAuth0(auto0Options)], schema);
```
Comment (n1ru4l) [inline packages/core/test/plugins/use-masked-errors.spec.ts]: the useMaskedErrors plugin is missing here

Comment (n1ru4l) [reply packages/core/test/plugins/use-masked-errors.spec.ts]: nevermind :D

**Review by dotansimha — APPROVED** [review]

**Pull request merged**